### PR TITLE
FIX: ensure we dont collapse data multiple times

### PR DESCRIPTION
**Pull request by jjaffeux**
Note that this commit will also disable daily grouping for datasets with more than 30 data points. This will also smartly do the grouping by month when grouping a full year.

### CODE REVIEW

**Review by AndrewPrigorshnev — COMMENTED** [review]
In general – a real good extracting of logic, it makes things clearer 👍.

File: app/assets/javascripts/admin/addon/models/report.js [hunk 1]
```diff
@@ -504,6 +504,63 @@ const Report = EmberObject.extend({
...
+      let currentEnd = startMoment.clone().endOf(isoKind);
+      const transformedData = [
+        {
+          x: currentStart.format("YYYY-MM-DD"),
+          y: 0,
+        },
+      ];
+
+      data.forEach((d) => {
+        let date = moment(d.x, "YYYY-MM-DD");
```
Comment (AndrewPrigorshnev) [inline app/assets/javascripts/admin/addon/models/report.js:536]:
```suggestion
        const date = moment(d.x, "YYYY-MM-DD");
```

File: app/assets/javascripts/admin/addon/models/report.js [hunk 2]
```diff
@@ -503,7 +503,78 @@ const Report = EmberObject.extend({
   },
 });
 
+function computeGrouping(data, grouping) {
+  if (!grouping) {
```
Comment (eviltrout) [inline app/assets/javascripts/admin/addon/models/report.js:507]: This seems to share some code with `_unitForDatapoints` above. Could it be DRYed up?

**Review by eviltrout — APPROVED** [review]

File: app/assets/javascripts/admin/addon/models/report.js (hunk 2 above)
Comment (jjaffeux) [reply app/assets/javascripts/admin/addon/models/report.js:507]: It mostly share the constants, so yes maybe I could use constants

**Comment by AndrewPrigorshnev** [comment]
LGTM 👍

**Review by AndrewPrigorshnev — APPROVED** [review]

**Pull request merged**